lib/active_campaign/v3/client.rb

```diff
@@ -54,16 +54,18 @@ def request(action, payload)
 
         case response.code
         when 403
-          {
-            'error' => 'The request could not be authenticated or the authenticated user is not authorized to access the requested resource.'
-          }
+          raise ActiveCampaign::Error, 'The request could not be authenticated or the authenticated user is not authorized to access the requested resource.'
         when 404
-          {
-            'error' => 'The requested resource does not exist.'
-          }
+          raise ActiveCampaign::Error, 'The requested resource does not exist.'
```

Comment:
It would probably be smart to subclass this error class for these specific types. I could see wanting to do something differently with a 404 vs. a 500 so being able to rescue them individually could be useful.

I would subclass them though from this class so it's still easy to rescue all errors in one `rescue` clause.

You might also want to include more info (maybe the whole response object) on the exception itself for debugging or error handling purposes.